httpclient-cache/src/main/java/org/apache/http/impl/client/cache/CacheKeyGenerator.java

```diff
@@ -63,45 +114,23 @@
      * @return String the extracted URI
      */
     public String getURI(final HttpHost host, final HttpRequest req) {
-        if (isRelativeRequest(req)) {
-            return canonicalizeUri(String.format("%s%s", host.toString(), req.getRequestLine().getUri()));
+        try {
+            final URI uri = normalize(getRequestUri(req, host));
```

Comment:
Why `uri` local var?

Reply:
@cstamas No reason. Just old habit. It is not a deal breaker, is it?

Reply:
Nope, is fine, I just found it strange 😄

Reply:
@cstamas I cant help it. `return normalize(getRequestUri(req, host)).toASCIIString();` just looks ugly to me.

Reply:
Agreed